### PR TITLE
[MB-15162] Fix various pre-commit issues

### PR DESCRIPTION
**Pull request by reggieriser**
## Summary

I noticed at some point that `pre-commit` started warning me about some fields being regex vs. glob:
```
[WARNING] The 'files' field in hook 'swagger' is a regex, not a glob -- matching '/*' probably isn't what you want here
[WARNING] The 'exclude' field in hook 'swagger' is a regex, not a glob -- matching '/*' probably isn't what you want here
```
This may have started when I picked up pre-commit 3.x locally when updating my brew dependencies.  This PR fixes that issue.  Note: I've rolled back to pre-commit 2.x locally for now because I was having runaway `golangci-lint` memory usage with pre-commit 3.x.

I also updated the `markdownlint-cli` hook to the latest version.  Note that I'm holding off on updating `golangci-lint` to a later version because there [is an issue with the gci linter](https://github.com/daixiang0/gci/issues/135#issuecomment-1419665501) that is blocking us (it moves and then deletes a linter suppression comment which doesn't appear easy to work around due to other expectations we have on placement of these comments).

When I was playing with the newer golangci-lint, it did point out an issue with one set of tests where we were ignoring a return value from an `Equal` call as opposed to doing a `suite.Equals`.  I went ahead and fixed those so we don't have to deal with them later.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

Just make sure all tests pass and pre-commit still works locally as expected.

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/main/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations)
- [ ] Have been communicated to #g-database
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -115,11 +115,11 @@ repos:
         name: Swagger
         entry: scripts/pre-commit-swagger-validate
         language: script
-        files: swagger/*
+        files: swagger/.*
```
Comment (reggieriser) [inline .pre-commit-config.yaml:118]: Warning I was seeing:
`[WARNING] The 'files' field in hook 'swagger' is a regex, not a glob -- matching '/*' probably isn't what you want here`

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -115,11 +115,11 @@ repos:
...
         types: [yaml]
-        exclude: swagger-def/* # These are partial swagger files that are compiled into the ones in swagger/*.
-                               # They will be checked unless explicitly excluded, but they will fail.
-                               # By only validating the compiled files, we are effectively checking these files as well.
+        exclude: swagger-def/.* # These are partial swagger files that are compiled into the ones in swagger/*.
```
Comment (reggieriser) [inline .pre-commit-config.yaml:120]: Warning I was seeing:
`[WARNING] The 'exclude' field in hook 'swagger' is a regex, not a glob -- matching '/*' probably isn't what you want here`

File: pkg/storage/storage.go [hunk 3]
```diff
@@ -1,7 +1,7 @@
 package storage
 
 import (
-	//RA Summary: gosec - G401 - Weak cryptographic hash
+	//RA Summary: gosec - G501 - Weak cryptographic hash
```
Comment (reggieriser) [inline pkg/storage/storage.go:4]: This is just a case where we had a mismatched linter ID here when compared to the `#nosec` suppression below it.

File: pkg/services/ppmshipment/ppm_shipment_updater_test.go [hunk 4]
```diff
@@ -109,7 +111,7 @@ func (suite *PPMShipmentSuite) TestUpdatePPMShipment() {
 		suite.NilOrNoVerrs(err)
 
 		// Fields that should now be updated
-		newPPM.ExpectedDepartureDate.Equal(updatedPPM.ExpectedDepartureDate)
+		suite.Equal(newPPM.ExpectedDepartureDate.Format(dateOnly), updatedPPM.ExpectedDepartureDate.Format(dateOnly))
```
Comment (reggieriser) [inline pkg/services/ppmshipment/ppm_shipment_updater_test.go:114]: Linter issue:

In a nutshell, this line was not testing anything.

I first changed this to do a `suite.Equal` between the two dates, but I was having time zone discrepancies -- the locally created date had timezone name `UTC` but one read from the DB that it's comparing against had an empty string for the timezone name which caused a diff (even though I believe both refer to UTC).  The fields we're comparing represent dates, not timestamps, so I just decided to check the string representations of the dates instead.

**Review by mandaem — APPROVED** [review]
LGTM

**Review by ahobson — APPROVED** [review]
🙇

File: .pre-commit-config.yaml (hunk 1 above)
Comment (rogeruiz) [reply .pre-commit-config.yaml:118]: I saw this too! thank you for confirming and fixing it.

**Review by rogeruiz — APPROVED** [review]
lgtm 🌈

> approving as IS3

**Pull request merged**